libcudacxx/include/cuda/__memory_resource/cuda_api_wrapper.h

```diff
@@ -0,0 +1,55 @@
+//===----------------------------------------------------------------------===//
```

Comment:
Move this to `cuda/std/detail` and drop references to MR in header guard, as this will be far more widely useful than just here, and this way we won't have to remember to move it around when more uses arise. (`stream_ref` should also get refactored to use these, but that's definitely a follow-up PR thing.)

Reply:
I am moving them to `cuda/std/__cuda`

I really want to get rid of the plain `detail` folder